### PR TITLE
docs: add fr translations for `architecture`, `getting-started`, `index`, `testing`, and `why-bloc`

### PR DESCRIPTION
**Pull request by Badlix**
## Status

**READY**

## Breaking Changes

NO

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [x] 📝 Documentation
- [ ] 🗑️ Chore

## Détails

Add the french translation for : 
- index.mdx
- getting-started.mdx
- why-bloc.mdx
- architecture.mdx
- testing.mdx

### CODE REVIEW

File: docs/src/content/docs/fr/architecture.mdx [hunk 1]
```diff
@@ -0,0 +1,120 @@
+---
+title: Architecture
+description: Vue d'ensemble des modèles d'architecture recommandés pour l'utilisation de Bloc.
+---
+
+#############
+### FINI 
+#############
```
Comment (felangel) [inline docs/src/content/docs/fr/architecture.mdx:8]:
```suggestion
```

File: docs/src/content/docs/fr/getting-started.mdx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+---
+title: Démarrer
+description: Tout le nécessaire pour débuter avec Bloc.
+---
+
+#############
+### FINI 
+#############
```
Comment (felangel) [inline docs/src/content/docs/fr/getting-started.mdx:8]:
```suggestion
```

File: docs/src/content/docs/fr/testing.mdx [hunk 3]
```diff
@@ -0,0 +1,68 @@
+---
+title: Tester
+description: Les bases de l'écriture de tests pour vos blocs.
+---
+
+#############
+### FINI 
+#############
```
Comment (felangel) [inline docs/src/content/docs/fr/testing.mdx:8]:
```suggestion
```

File: docs/src/content/docs/fr/why-bloc.mdx [hunk 4]
```diff
@@ -0,0 +1,35 @@
+---
+title: Pourquoi utiliser Bloc?
+description: Un aperçu de ce qui fait de Bloc une solution solide pour la gestion d'état.
+sidebar:
+  order: 1
+---
+
+#############
+### FINI 
+#############
```
Comment (felangel) [inline docs/src/content/docs/fr/why-bloc.mdx:10]:
```suggestion
```

**Comment by felangel** [comment]
Looks like we just need to run `npm run format` to fix the formatting.

File: docs/src/content/docs/fr/testing.mdx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+Pour rappel, l'implémentation du `CounterBloc` ressemble à ça :
+
+<CounterBlocSnippet />
+
+## Configuration
+
+Avant de commencer à écrire nos tests, nous avons besoins d'ajouter des framework de test dans nos dépendances,
+
+Nous allons ajouter [test](https://pub.dev/packages/test) et [bloc_test](https://pub.dev/packages/bloc_test) à notre projet.
```
Comment (vmichalak) [inline docs/src/content/docs/fr/testing.mdx:27]: **suggestions:** "ajouter **les packages** [test] ..." to be more clear

File: docs/src/content/docs/fr/why-bloc.mdx [hunk 6]
```diff
@@ -0,0 +1,32 @@
+---
+title: Pourquoi utiliser Bloc?
+description: Un aperçu de ce qui fait de Bloc une solution solide pour la gestion d'état.
+sidebar:
+  order: 1
+---
+
+
+Bloc permet de facilement séparer les composants visuels de la logigue métier, rendant votre code `rapide`, `facile à tester` et `réutilisable`.
```
Comment (vmichalak) [inline docs/src/content/docs/fr/why-bloc.mdx:9]: **issue:** **logique**

File: docs/src/content/docs/fr/why-bloc.mdx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+Lorsqu'on développe des application de qualité destinés à être utilisé en production, la gestion des états devient critique.
```
Comment (vmichalak) [inline docs/src/content/docs/fr/why-bloc.mdx:11]: **issue:** des applications de qualité **destinées** à

File: docs/src/content/docs/fr/why-bloc.mdx [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+En tant que développeurs, nous souhaitons :
+
+- connaître l'état de notre application à tout moment.
+- tester facilement chaque cas pour nous assurer que notre application réagit correctement.
+- record every single user interaction in our application so that we can make data-driven decisions.
```
Comment (vmichalak) [inline docs/src/content/docs/fr/why-bloc.mdx:17]: **issue: enregistrer chaque interaction utilisateur dans notre application**

File: docs/src/content/docs/fr/why-bloc.mdx [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+- travailler de manière aussi efficace que possible et réutiliser des composants au sein de notre application ainsi que dans des applications différentes.
+- permettre à de nombreux développeurs de travailler en harmonie en suivant des modèles et des conventions communes.
+- développer des applications rapides et réactives.
+
+Bloc a été conçu pour répondre à tous ces besoins et bien plus encore.
+
+Il existe beaucoup de solution pour gérer les états et décider laquelle utiliser peut être une tâche laborieuse. Il n'y a pas de solution parfaite! Ce qui est important est de choisir celle qui correspond le mieux à ton équipe et à ton projet.
```
Comment (vmichalak) [inline docs/src/content/docs/fr/why-bloc.mdx:24]: **issue:** beaucoup de **solutions** pour gérer les états

File: docs/src/content/docs/fr/why-bloc.mdx [hunk 10]
```diff
@@ -0,0 +1,32 @@
+---
+title: Pourquoi utiliser Bloc?
+description: Un aperçu de ce qui fait de Bloc une solution solide pour la gestion d'état.
```
Comment (vmichalak) [inline docs/src/content/docs/fr/why-bloc.mdx:3]: **issue:** une solution solide pour la gestion **d’états**

File: docs/src/content/docs/fr/why-bloc.mdx [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+- record every single user interaction in our application so that we can make data-driven decisions.
+- travailler de manière aussi efficace que possible et réutiliser des composants au sein de notre application ainsi que dans des applications différentes.
+- permettre à de nombreux développeurs de travailler en harmonie en suivant des modèles et des conventions communes.
+- développer des applications rapides et réactives.
+
+Bloc a été conçu pour répondre à tous ces besoins et bien plus encore.
+
+Il existe beaucoup de solution pour gérer les états et décider laquelle utiliser peut être une tâche laborieuse. Il n'y a pas de solution parfaite! Ce qui est important est de choisir celle qui correspond le mieux à ton équipe et à ton projet.                
+
+Bloc a été conçu avec trois principes fondamentales en tête :
```
Comment (vmichalak) [inline docs/src/content/docs/fr/why-bloc.mdx:26]: **issue:** trois principes **fondamentaux**

File: docs/src/content/docs/fr/why-bloc.mdx [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+
+- **Simple:** Facile à comprendre et utilisable par des développeurs de niveaux variés.
+- **Puissant:** Aide à faire des applications complexes en les assemblant à partir de composants plus simple.
```
Comment (vmichalak) [inline docs/src/content/docs/fr/why-bloc.mdx:29]: **issue:** composants plus **simples**

File: docs/src/content/docs/fr/why-bloc.mdx [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+Lorsqu'on développe des application de qualité destinés à être utilisé en production, la gestion des états devient critique.
+
+En tant que développeurs, nous souhaitons :
+
+- connaître l'état de notre application à tout moment.
+- tester facilement chaque cas pour nous assurer que notre application réagit correctement.
+- record every single user interaction in our application so that we can make data-driven decisions.
+- travailler de manière aussi efficace que possible et réutiliser des composants au sein de notre application ainsi que dans des applications différentes.
+- permettre à de nombreux développeurs de travailler en harmonie en suivant des modèles et des conventions communes.
+- développer des applications rapides et réactives.
```
Comment (vmichalak) [inline docs/src/content/docs/fr/why-bloc.mdx:20]: **suggestion:** développer des applications **performantes** et réactives

**Review by vmichalak — COMMENTED** [review]
Thanks a lot for your contribution, it's a lot of work to translate the documentation.

Since I am French, I took the liberty of highlighting the typos i've see in your PR. All comments marked as suggestion are more to make the documentation more "french native" / "natural" to read in french. Don't hesitate to comment in case of doubt :)

**Review by felangel — APPROVED** [review]
LGTM thanks so much for the contributions and @vmichalak for the review I really appreciate it 💙 🙏

**Pull request merged**